interface/main/tabs/main.php

```diff
@@ -510,5 +509,31 @@ function jsFetchGlobals(scope) {
 document.addEventListener('touchstart', {}); //specifically added for iOS devices, especially in iframes
 </script>
 
+<script>
+    // set up global translations for js
+    function setupI18n(lang_id) {
```

Comment:
I'd move this to top of main where I have jsFetchGlobals and remove and replace with this routine. I'd also have setupI18n(lang_id) return a promise(similar to jsFetchGlobals) then init just below. By setupI18n(lang_id)  returning a promise then it can be re-init'ed elsewhere if needed. At same time a new global function:
`var xl = function (string) {
   top.i18next.t(string)
}`
Could be added.

Reply:
will do